soccer/gameplay/plays/testing/four_courners_pass.py

```diff
@@ -0,0 +1,229 @@
+							'immediately')
+		#transition when ball is captured
+		self.add_transition(
+			FourCornerPass.State.setup, FourCornerPass.State.passing, lambda: self.is_setup_done() , 'all subbehaviors completed')
+
+		#transition when ball is passed or failed to pass
+		self.add_transition(
+			FourCornerPass.State.passing, FourCornerPass.State.setup, lambda: self.subbehavior_with_name('passer').state == behavior.Behavior.
+			State.completed or self.subbehavior_with_name('passer').state == behavior.Behavior.
+			State.failed, 'all subbehaviors completed')
+
+		#change this to adjust the square size
+		variable_square = 4.5
+
+		#NEVER CHANGE THIS
+		length = constants.Field.Length
+		width = constants.Field.Width
+
+		#if the square size is smaller than the width shrink it down otherwise
+		#max value is the width of the field.
+		if (variable_square < width) :
+			width = variable_square
+
+		#radius of robot
+		self.bot = constants.Robot.Radius
+		# the largest and smallest x and y postion possible for the square
+		self.max_x = (width / 2 - self.bot * 3)
+		self.min_x = -self.max_x
+		self.max_y = length/2 + self.max_x
+		self.min_y = length/2 - self.max_x
+
+		# the four courners
+		self.square_points = [
+			robocup.Point(self.max_x, self.max_y),
+			robocup.Point(self.min_x, self.max_y),
+			robocup.Point(self.min_x, self.min_y),
+			robocup.Point(self.max_x, self.min_y)]
+
+		# speed of the hunting robots
+		self.normal_speed = 2.0
+		# speed of the defending robots can decrease value to make it easier for offense
+		self.defense_speed = variable_square/3.0
+
+		# picks the direction to pass to. TODO make actual smart pass selection
+		self.direction = 1
+
+	# constanly running changes and updates
+	def execute_running(self):
+		# checks if there is an offensive behavior for the various plays and set the normal speed
+		if (self.has_subbehavior_with_name('move1')) :
+			robo = self.subbehavior_with_name('move1').robot
+			if (robo != None) :
+				robo.set_max_speed(self.normal_speed)
+
+		if (self.has_subbehavior_with_name('move2')) :
+			robo = self.subbehavior_with_name('move2').robot
+			if (robo != None) :
+				robo.set_max_speed(self.normal_speed)
+
+		if (self.has_subbehavior_with_name('capture')):
+			robo = self.subbehavior_with_name('capture').robot
+			if (robo != None) :
+				robo.set_max_speed(self.normal_speed)
+
+		if (self.has_subbehavior_with_name('premove')) :
+			robo = self.subbehavior_with_name('premove').robot
+			if (robo != None) :
+				robo.set_max_speed(self.normal_speed)
+
+		# checks if there is a chasing robot and set their speed to defense speed
+		if (self.has_subbehavior_with_name('Chasing')):
+			robo = self.subbehavior_with_name('Chasing').robot
+			if (robo != None) :
+				robo.set_max_speed(self.defense_speed)
+
+		# keep the iso point stuck in the corner
+		if (self.has_subbehavior_with_name('Iso')) :
+			self.iso.pos = robocup.Point(-constants.Field.Width/2, 0)
```

Comment:
iso?

Reply:
Iso is isolating a robot in the corner so he can do nothing.

Reply:
This can probably be removed since the subbehavior will always try to move to a specific point. You shouldn't need to keep setting the target point

Reply:
Even if it's completed?

Reply:
Move should always go from completed -> running if the robot gets bumped too far away from the target point

Reply:
It's removed